plugins/urls/handlers/website.py

```diff
@@ -0,0 +1,266 @@
+            "Headers: {0}", list(response.headers)
+        )
+        self.plugin.logger.trace("HTTP code: {0}", response.status_code)
+
+        if "content-type" not in response.headers:
+            response.headers["Content-Type"] = ""
+
+        content_type = response.headers["content-type"].lower()
+
+        if ";" in content_type:
+            content_type = content_type.split(";")[0]
+
+        if content_type not in context["config"]["content_types"]:
+            self.plugin.logger.debug(
+                "Unsupported Content-Type: %s"
+                % response.headers["content-type"]
+            )
+            return  # Not a supported content-type
+
+        soup = BeautifulSoup(response.text)
+
+        if soup.title and soup.title.string:
+            title = soup.title.string.strip()
+            title = re.sub("\s+", " ", title)
+            title = title
+
+            context["event"].target.respond(
+                to_unicode('"{0}" at {1}'.format(
+                    to_bytes(title), to_bytes(url.domain)
+                ))
+            )
+        else:
+            self.plugin.logger.debug("No title")
+
+        if session.session_type:
+            session.cookies.save(ignore_discard=True)
+
+            if len(session.cookies):
+                session.cookies.file_exists = True
+                session.cookies.load()
+
+    def errback(self, error, url, context, session):
+        self.plugin.logger.error("Error parsing URL")
+
+        if isinstance(error.value, ResponseNeverReceived):
+            for f in error.value.reasons:
+                f.printDetailedTraceback()
+                context["event"].target.respond(
+                    '"{0}" at {1}'.format(f.getErrorMessage(), url.domain)
+                )
+        else:
+            context["event"].target.respond(
+                '"{0}" at {1}'.format(error.getErrorMessage(), url.domain)
+            )
+            error.printDetailedTraceback()
+
+        if session.session_type:
+            session.cookies.save(ignore_discard=True)
+
+            if len(session.cookies):
+                session.cookies.file_exists = True
+                session.cookies.load()
+
+    def check_blacklist(self, url, context):
+        for entry in context["config"]["blacklist"]:
+            if re.match(entry, url, flags=str_to_regex_flags("ui")):
+                self.urls_plugin.logger.debug(
+                    "Matched blacklist regex: %s" % entry
+                )
+                return True
+
+        return False
+
+    def get_cookie_jar(self, filename):
+        cj = ChocolateCookieJar(self.cookies_base_path + filename)
+
+        try:
+            cj.load()
+        except LoadError as e:
+            self.plugin.logger.exception(
+                "Failed to load cookie jar {0}".format(filename)
+            )
+        except IOError as e:
+            self.plugin.logger.debug(
+                "Failed to load cookie jar {0}: {1}".format(filename, e)
```

Comment:
Why isn't this used for the `file_exists` thing below? Why does an empty cookie jar count as file not existing? Also, where is `file_exists` defined? This is the only reference I can find of it.